### ISSUE TITLE
Update PUT verb for services

### ISSUE THREAD

**Issue by waltermoreira** [Arabidopsis-Information-Portal/adama#70]
Code for handling PUT has gone out-of-date (b/c of duplication of code).  Follow DRY and use the same code to specify the fields for creation and updating.

**Comment by vivekkrish** [comment]
Are there any plans to support the HTTP PATCH verb, which could incrementally (delta) update an existing namespace and/or adapter metadata? (http://restful-api-design.readthedocs.org/en/latest/methods.html#patch-vs-put)

**Comment by waltermoreira** [comment]
It's a possibility, indeed.  However, I'm experimenting with a restructure of the registration procedure (and other things) in the branch `async_registration`, and I was thinking to deprecate the big number of parameters in favor of a smaller subset together with the `metadata.yml`.  If we end up following this path, then a PATCH verb would not have the same possible set of parameters as a POST or PUT.

It could still make sense, as a way to update fields on the metadata without rebuilding the services, but it also means that the file `metadata.yml` and the actual metadata of the service can go out of sync.

I'm open to discussion, anyway.

**Comment by mwvaughn** [comment]
Is it plausible to simply allow re-POST without going through the delete cycle? This is the access pattern Agave services uses for apps and systems.

**Comment by waltermoreira** [comment]
Yes, we could support this, i.e. issuing an implicit DELETE when POSTing to an existing service.  I'll take a stab at this and push to the dev instance.
